src/common/nfts/data/data-sources/nft.mongo.source.ts

```diff
@@ -0,0 +1,36 @@
+import { MongoSource } from '@core/storage/data-sources/mongo.source';
+import { DataSourceOperationError } from '@core/architecture/data/errors/data-source-operation.error';
+import { NftDocument } from '../nfts.dtos';
+import { CollectionMongoSource } from '@core/storage/data-sources/collection.mongo.source';
+
+/**
+ * NFTs data source from the mongo database
+ * @class
+ */
+export class NftMongoSource extends CollectionMongoSource<NftDocument> {
+  public static Token = 'NFT_MONGO_SOURCE';
+
+  /**
+   * @constructor
+   * @param {MongoSource} mongoSource
+   */
+  constructor(mongoSource: MongoSource) {
+    super(mongoSource, 'nfts');
+  }
+
+  /**
+   * Insert multiple documents in one set.
+   *
+   * @async
+   * @returns {string[]} IDs of added documents
+   * @throws {DataSourceWriteError}
+   */
+  public async insertOne(dto: NftDocument): Promise<string> {
```

Comment:
Result<string> ?

Reply:
Results are returned by repositories and use-cases/services... sources should throw errors or handle these which are not important for the business logic...